examples/example_obj_lib.py

```diff
@@ -0,0 +1,130 @@
+
+
+_file_map = {'monthly':
+             '../aospy/test/data/netcdf/000[4-6]0101.precip_monthly.nc'}
+example_run = Run(
+    name='example_run',
+    description=(
+        'Control simulation of the idealized moist model'
+    ),
+    default_start_date=datetime.datetime(4, 1, 1),
+    default_end_date=datetime.datetime(6, 12, 31),
+    data_loader=DictDataLoader(_file_map)
+)
+
+
+example_model = Model(
+    name='example_model',
+    grid_file_paths=(
+        '../aospy/test/data/netcdf/00040101.precip_monthly.nc',
+        '../aospy/test/data/netcdf/im.landmask.nc'
+    ),
+    runs=[example_run]
+)
+
+
+precip_largescale = Var(
+   name='precip_largescale',
+   alt_names=('condensation_rain',),
+   def_time=True,
+   description='Precipitation generated via grid-scale condensation',
+)
+
+
+precip_convective = Var(
+   name='precip_convective',
+   alt_names=('convection_rain',),
+   def_time=True,
+   description='Precipitation generated by the convective parameterization',
+)
+
+
+def total_precipitation(precip_largescale, precip_convective):
+    """Sum of convective and large-scale precipitation.
+
+    Parameters
+    ----------
+    precip_largescale, precip_convective : xarray.DataArrays
+        Precipitation from grid-scale condensation and from convective
+        parameterization, respectively.
+
+    Returns
+    -------
+    xarray.DataArray
+    """
+    return precip_largescale + precip_convective
+
+
+def conv_precip_frac(precip_largescale, precip_convective):
+    """Fraction of total precip that is from convection parameterization.
+
+    Parameters
+    ----------
+    precip_largescale, precip_convective : xarray.DataArrays
+        Precipitation from grid-scale condensation and from convective
+        parameterization, respectively.
+
+    Returns
+    -------
+    xarray.DataArray
+    """
+    total = total_precipitation(precip_largescale, precip_convective)
+    # Mask using xarray's `where` method to prevent divide-by-zero.
+    return precip_convective / total.where(total)
+
+
+precip_total = Var(
+   name='precip_total',
+   def_time=True,
+   func=total_precipitation,
+   variables=(precip_largescale, precip_convective),
+)
+
+
+precip_conv_frac = Var(
+   name='precip_conv_frac',
+   def_time=True,
+   func=conv_precip_frac,
+   variables=(precip_largescale, precip_convective),
+)
+
+
+globe = Region(
+    name='globe',
+    description='Entire globe',
+    lat_bounds=(-90, 90),
+    lon_bounds=(0, 360),
+    do_land_mask=False
+)
+
+
+tropics = Region(
+    name='tropics',
+    description='Tropics, defined as 30S-30N',
+    lat_bounds=(-30, 30),
+    lon_bounds=(0, 360),
+    do_land_mask=False
+)
+
+
+example_proj = Proj(
+    'example_proj',
+    direc_out='example-output',
+    tar_direc_out='example-tar-output',
+    models=(example_model,),
+    regions=(globe, tropics)
+)
+
+
+projects = {example_proj.name: example_proj}
+variables = {var.name: var for var in [precip_largescale, precip_convective,
```

Comment:
>this requirement (creating a dictionary mapping Var name attributes to Var objects) breaks my current object library, because I have some Var objects that have the same name attribute, but are different Var objects

>this practice of requiring the user to create module level dictionaries to set up their object libraries feels a little non-ideal

These are both good points and I definitely want to avoid breaking your workflow.  Will have a separate comment and/or Issue on these...I have some ideas but need to clarify them.